src/trans.c

```diff
@@ -1172,7 +1140,7 @@ Obj FuncIMAGE_SET_TRANS(Obj self, Obj f)
     Obj out = FuncUNSORTED_IMAGE_SET_TRANS(self, f);
 
     if (!IS_SSORT_LIST(out)) {
-        SORT_PLIST_CYC(out);
+        SortPlistByMasterPointer(out);
         RetypeBag(out, T_PLIST_CYC_SSORT + IMMUTABLE);
         CHANGED_BAG(out);
```

Comment:
Reminder to self: get rid of this useless `CHANGED_BAG` (and other similar ones below) in a future PR (or feel free to remove it now; but no need to bloat this PR)